web/profiles/custom/yalesites_profile/modules/custom/ys_views_basic/src/ViewsBasicManager.php

```diff
@@ -0,0 +1,147 @@
+   *
+   * @var array
+   */
+  const ALLOWED_ENTITIES = [
+    'node' => 'node_type',
+    // 'media' => 'media_type',
+  ];
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity display repository.
+   *
+   * @var \Drupal\Core\Entity\EntityDisplayRepository
+   */
+  protected $entityDisplayRepository;
+
+  /**
+   * Constructs a new ViewsBasicManager object.
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entity_type_manager,
+    EntityDisplayRepository $entity_display_repository
+    ) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityDisplayRepository = $entity_display_repository;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('entity_display.repository')
+    );
+  }
+
+  /**
+   * Returns an array of entity type machine names and the human readable name.
+   */
+  public function entityTypeList() {
+    $entityTypes = [];
+
+    foreach (self::ALLOWED_ENTITIES as $type) {
+      $types = $this->entityTypeManager()
+        ->getStorage($type)
+        ->loadMultiple();
+      foreach ($types as $machine_name => $object) {
+        $nodeType = $object->getEntityTypeId();
+        $value = "$nodeType.$machine_name";
+        $entityTypes[$value] = $object->label();
+      }
+    }
+
+    return $entityTypes;
+  }
+
+  /**
+   * Returns an array of view mode machine names and the human readable name.
+   */
+  public function viewModeList() {
+    $viewModes = [];
+
+    $view_modes = $this->entityTypeManager()
+      ->getStorage('entity_view_mode')
+      ->loadMultiple();
+    foreach ($view_modes as $machine_name => $object) {
+      foreach (self::ALLOWED_ENTITIES as $key => $type) {
+        $pattern = "/^{$key}/";
+        if (preg_match($pattern, $machine_name) && $object->status()) {
+          $viewModes[$machine_name] = $object->label();
+        }
+      }
+    }
+
+    return $viewModes;
+  }
+
+  /**
+   * Returns an entity label given an entity type and machine name.
+   */
+  public function getEntityLabel($type) {
+    $entityInfo = explode(".", $type);
+    $bundleLabel = $this->entityTypeManager
+      ->getStorage($entityInfo[0])
+      ->load($entityInfo[1])
+      ->label();
+    return $bundleLabel;
+  }
+
+  /**
+   * Returns a view mode label given an view mode type stored in the params.
+   */
+  public function getViewModeLabel($type) {
+    $viewModeInfo = explode(".", $type);
+    return $this->entityDisplayRepository
+      ->getViewModes($viewModeInfo[0])[$viewModeInfo[1]]['label'];
+  }
+
+  /**
+   * Returns a default value for a parameter to auto-select one in the list.
+   */
+  public function getDefaultParamValue($type, $params) {
+    $paramsDecoded = json_decode($params, TRUE);
```

Comment:
You are encoding and decoding the params. But you also have the field set to 'serialize'. I believe this is redundant. If you are serializinng the field, then you can simply access the data as an array. No need to encode or decode it.

Reply:
Since the field is set to `'type' => 'text',` I tried simply setting it as the array and thought it would serialize it for me, but it complained that it was a type array instead of text. So, I added the `serialize()` and `unserialize()` functions. Let me know if you think that will work for our needs.